### PR TITLE
Simplify openpbs job states

### PR DESCRIPTION
**Pull request by eivindjahren**
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/ert/scheduler/openpbs_driver.py [hunk 1]
```diff
@@ -30,7 +29,7 @@
 logger = logging.getLogger(__name__)
 
 _POLL_PERIOD = 2.0  # seconds
-JobState = Literal[
+JOB_STATES = [
```
Comment (xjules) [inline src/ert/scheduler/openpbs_driver.py:32]: The issue is that we have the same pattern; ie. Literal states in other drivers too.

Comment (eivindjahren) [reply src/ert/scheduler/openpbs_driver.py:32]: This was only used to iterate over in the openpbs driver and in the test so it didn't need to be a Literal, just a list. The other ones seem to be also used for typing.

Comment (xjules) [reply src/ert/scheduler/openpbs_driver.py:32]: Seems to be fine though.

**Review by xjules — APPROVED** [review]
Let's ship it! :rocket:

**Pull request merged**